app/controllers/posts_controller.rb

```diff
@@ -345,7 +345,9 @@ def destroy_many
     posts.each { |p| guardian.ensure_can_delete!(p) }
 
     Post.transaction do
-      posts.each { |p| PostDestroyer.new(current_user, p, defer_flags: defer_flags).destroy }
+      posts.each_with_index do |p, i|
+        PostDestroyer.new(current_user, p, defer_flags: (agree_with_first_reply_flag.to_s == "true" && i == 0) ? "false" : "true").destroy
```

Comment:
It's a bit odd that we supply a string based boolean here to `PostDestroyer`.

I'm ok with the parameter to the controller being "true" but `PostDestoyer` should take `true`. Would you consider making that change? I bet it would not be significant.